src/rust/engine/fs/src/lib.rs

```diff
@@ -648,6 +731,52 @@ impl Vfs<String> for DigestTrie {
         )))
     }
 
+    async fn path_metadata(&self, path: PathBuf) -> Result<Option<PathMetadata>, String> {
+        let entry = match self.entry(&path)? {
+            Some(e) => e,
+            None => return Ok(None),
+        };
+
+        Ok(Some(match entry {
+            directory::Entry::File(f) => PathMetadata {
+                path: PathBuf::from(f.name().as_str()),
```

Comment:
It looks like this might be just the basename, not the full path, while the "real FS" implementation above uses the full path. (i.e. if given `foo/bar/baz.txt`, that one will set this to that full path, while this will store just `baz.txt`?)

Have I misunderstood?

Reply:
Looking into the code, I believe you are correct. Then this logic should just propagate `path` and not query the entry for its `name`.

Reply:
Fixed.